### DISCUSSION TITLE
Migrating react-icons from 2.2.7 to 4.10.1

### DISCUSSION THREAD

**Discussion by kirtanchandak** [react-icons/react-icons#774] [General]
Hey Icons, 

### I am migrating my code base from react-icons version 2.2.7 to 4.10.1 and I have so many imports like - 
```
react-icons/lib/io/android-locate
react-icons/lib/io/chevron-right
```

### Any help would be appreciated 🙏

**Comment by joadr** [comment]
I created a script to do this, it correctly identifies import lines that follow the v2 pattern, and transforms them into the v4 pattern.

Here's how the script works:

1. **fromDir** is a recursive function that finds all JavaScript files in a directory and its subdirectories.

2. **toPascalCase** is a function that turns kebab-case strings into PascalCase.

3. For each JavaScript file, you read its content, and for each match of the regular expression (which represents an old-style import statement), you replace it with a new-style import statement.

4. You store each old-style imported object name along with its new-style equivalent in the `substitutions` object.

5. You then iterate over the `substitutions` object, and for each key-value pair, replace all instances of the key in the file content with its corresponding value.

6. Finally, you overwrite the file with the new content.


---------------

run `node migrate.js`

```js
// migrate.js
const fs = require('fs')
const path = require('path')

function fromDir(startPath, filter, callback) {
  if (!fs.existsSync(startPath)) {
    console.log('No directory found: ', startPath)
    return
  }

  let files = fs.readdirSync(startPath)
  for (let i = 0; i < files.length; i++) {
    let filename = path.join(startPath, files[i])
    let stat = fs.lstatSync(filename)
    if (stat.isDirectory()) {
      fromDir(filename, filter, callback)
    } else if (filter.test(filename)) callback(filename)
  }
}

function toPascalCase(str) {
  return str
    .split('-')
    .map(s => s.charAt(0).toUpperCase() + s.slice(1))
    .join('')
}

fromDir('./', /\.js$/, function (filename) {
  let content = fs.readFileSync(filename, 'utf8')
  const substitutions = {}
  let newContent = content.replace(
    /import ([A-Za-z]+) from 'react-icons\/lib\/([a-z]+)\/([a-z-]+)'/g,
    function (_, p1, p2, p3) {
      let newP2 = p2.charAt(0).toUpperCase() + p2.slice(1)
      let newP3 = toPascalCase(p3)

      // Update import ocurrences
      substitutions[p1] = `${newP2}${newP3}`
      return `import { ${p1} } from 'react-icons/${p2}'`
    }
  )

  // Update component ocurrences
  Object.keys(substitutions).forEach(key => {
    newContent = newContent.replace(new RegExp(key, 'gms'), substitutions[key])
  })

  fs.writeFileSync(filename, newContent, 'utf8')
})

console.log('Updated imports')
```
